flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -397,7 +398,12 @@ public static String getTaskManagerShellCommand(
 
 		startCommandValues.put("jvmmem", StringUtils.join(params, ' '));
 
-		String javaOpts = flinkConfig.getString(CoreOptions.FLINK_JVM_OPTIONS);
+		String javaOpts = "";
+		// Add default gc logging options if enabled
+		javaOpts += getGCLoggingOpts(logDirectory, flinkConfig);
+		// Add default heap dump options if enabled
+		javaOpts += " " + getHeapdumpOpts(appId, "taskmanager", logDirectory, flinkConfig);
```

Comment:
What if `getGCLoggingOpts` returns `""`? Then we would not need `" "`.

Reply:
Thank you for point it out! I will update.